lesson_2_pull/lesson_2_decorator.py

```diff
@@ -0,0 +1,82 @@
+"""
+    source of def object_memory(count):
+    https://habr.com/ru/post/193890/
+
+    source of LFUCache:
+    https://www.geeksforgeeks.org/cachetools-module-in-python/
+    https://github.com/rfyiambest/pylfu/blob/master/pylfu/pylfu.py
+
+"""
+import functools
+import sys
+
+# from cachetools import cached, LFUCache
+from collections import OrderedDict
+
+import requests
+
+
+def object_memory(count):
+    @functools.wraps(count)
+    def wrapper(*args, **kwargs):
+        result = count(*args, **kwargs)
+        print(f'Function: {count.__name__}, use: {sys.getsizeof(result)} bytes')
+        return result
+
+    return wrapper
+
+
+def key_func(x):
+    return x['freq']
+
+
+@object_memory
+def lfu_cache_decorator(max_limit=64):
+    def internal(f):
+        @functools.wraps(f)
+        def wrapper(*args, **kwargs):
+            cache_key = (*args, tuple(kwargs.items()))
+            if cache_key in wrapper._cache:
+                wrapper._cache[cache_key]['freq'] += 1
+                return wrapper._cache[cache_key]['value']
+            result = f(*args, **kwargs)
+            if len(wrapper._cache) >= max_limit:
+                min_freq = min(wrapper._cache.values(), key=key_func)['freq']
```

Comment:
remove key_func, you cant use external func in decorator, use lambda if you need function